x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -677,7 +718,10 @@ private ListenableFuture<ContentAndJwksAlgs> getFuture() {
 
                 final ListenableFuture<ContentAndJwksAlgs> newFuture = new ListenableFuture<>();
                 if (this.reloadFutureRef.compareAndSet(null, newFuture)) {
-                    loadInternal(ActionListener.runAfter(newFuture, () -> this.reloadFutureRef.compareAndSet(newFuture, null)));
+                    loadInternal(ActionListener.runAfter(newFuture, () -> {
+                        final ListenableFuture<ContentAndJwksAlgs> oldValue = this.reloadFutureRef.getAndSet(null);
+                        assert oldValue == newFuture : "future reference changed unexpectedly";
```

Comment:
This change is unlikely to make real difference. But I think it is safer to always set to null regardless just in case compare ever not match and result into the reference holding a stale future.